src/framework/ui/internal/keynavigationcontroller.cpp

```diff
@@ -304,21 +372,130 @@ void KeyNavigationController::prevSubSection()
     if (!activeSubSec) { // no any active
         IKeyNavigationSubSection* lastSub = lastEnabled(subsections);
         if (lastSub) {
-            lastSub->setActive(true);
+            activateSubSection(lastSub);
         }
         return;
     }
 
-    activeSubSec->setActive(false);
+    deactivateSubSection(activeSubSec);
 
     IKeyNavigationSubSection* prevSubSec = prevEnabled(subsections, activeSubSec);
     if (!prevSubSec) { // active is first
         IKeyNavigationSubSection* lastSub = lastEnabled(subsections);
         if (lastSub) {
-            lastSub->setActive(true);
+            activateSubSection(lastSub);
+        }
+        return;
+    }
+
+    activateSubSection(prevSubSec);
+}
+
+void KeyNavigationController::nextControl()
+{
+    LOGI() << "====";
+
+    const QList<IKeyNavigationControl*>& controls = controlsOfActiveSubSection();
+    if (controls.isEmpty()) {
+        return;
+    }
+
+    IKeyNavigationControl* activeControl = findActive(controls);
+    if (!activeControl) { // no any active
+        IKeyNavigationControl* first = firstEnabled(controls);
+        if (first) {
+            first->setActive(true);
+        }
+        return;
+    }
+
+    activeControl->setActive(false);
+
+    IKeyNavigationControl* nextControl = nextEnabled(controls, activeControl);
+    if (!nextControl) { // active is last
+        IKeyNavigationControl* first = firstEnabled(controls);
+        if (first) {
+            first->setActive(true);
+        }
+        return;
+    }
+
+    nextControl->setActive(true);
+}
+
+void KeyNavigationController::prevControl()
```

Comment:
I think a better name for this method is goToPrevControl (if I understood correctly)

Reply:
renamed